src/main.rs

```diff
@@ -99,22 +126,68 @@ fn main() {
         "--threshold must be between 1 and --shares"
     );
 
+    let format_type = match matches.value_of("format-type") {
+        Some(format_type_str) => format_type_str
+            .parse::<FormatType>()
+            .expect("--format-type must be one of the following: [hex|bip39]"),
+        None => FormatType::Hex,
+    };
+
+    let format_error = || panic!("bip39 encoding is only available for 128 and 256 bit keys");
+
     match bitsize {
```

Comment:
Here it would be more compact and more readable to swap the order of the matches, i.e. match on the `format_type` first from the outside, and then inside it match on the bitsize.